packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/utils/get-available-aggregations-from-object-fields.util.ts

```diff
@@ -12,101 +16,168 @@ export type AggregationField = {
-        aggregateOperation: AGGREGATE_OPERATIONS.min,
-      };
-
-      acc[`max${capitalize(field.name)}`] = {
-        type: GraphQLFloat,
-        description: `Maximum amount contained in the field ${field.name}`,
-        fromField: field.name,
-        aggregateOperation: AGGREGATE_OPERATIONS.max,
-      };
-
-      acc[`avg${capitalize(field.name)}`] = {
-        type: GraphQLFloat,
-        description: `Average amount contained in the field ${field.name}`,
-        fromField: field.name,
-        aggregateOperation: AGGREGATE_OPERATIONS.avg,
-      };
-
-      acc[`sum${capitalize(field.name)}`] = {
-        type: GraphQLFloat,
-        description: `Sum of amounts contained in the field ${field.name}`,
-        fromField: field.name,
-        aggregateOperation: AGGREGATE_OPERATIONS.sum,
-      };
-    }
-
-    if (field.type === FieldMetadataType.CURRENCY) {
-      acc[`min${capitalize(field.name)}AmountMicros`] = {
-        type: GraphQLFloat,
-        description: `Minimum amount contained in the field ${field.name}`,
-        fromField: field.name,
-        fromSubField: 'amountMicros',
-        aggregateOperation: AGGREGATE_OPERATIONS.min,
-      };
-
-      acc[`max${capitalize(field.name)}AmountMicros`] = {
-        type: GraphQLFloat,
-        description: `Maximal amount contained in the field ${field.name}`,
+      acc[`countNotEmpty${capitalize(columnName)}`] = {
+        type: GraphQLInt,
+        description: `Number of non-empty values for ${field.name}`,
         fromField: field.name,
-        fromSubField: 'amountMicros',
-        aggregateOperation: AGGREGATE_OPERATIONS.max,
+        fromFieldType: field.type,
+        fromSubField,
+        aggregateOperation: AGGREGATE_OPERATIONS.countNotEmpty,
       };
 
-      acc[`sum${capitalize(field.name)}AmountMicros`] = {
+      acc[`percentageEmpty${capitalize(columnName)}`] = {
         type: GraphQLFloat,
-        description: `Sum of amounts contained in the field ${field.name}`,
+        description: `Percentage of empty values for ${field.name}`,
         fromField: field.name,
-        fromSubField: 'amountMicros',
-        aggregateOperation: AGGREGATE_OPERATIONS.sum,
+        fromFieldType: field.type,
+        fromSubField,
+        aggregateOperation: AGGREGATE_OPERATIONS.percentageEmpty,
       };
 
-      acc[`avg${capitalize(field.name)}AmountMicros`] = {
+      acc[`percentageNotEmpty${capitalize(columnName)}`] = {
         type: GraphQLFloat,
-        description: `Average amount contained in the field ${field.name}`,
+        description: `Percentage of non-empty values for ${field.name}`,
         fromField: field.name,
-        fromSubField: 'amountMicros',
-        aggregateOperation: AGGREGATE_OPERATIONS.avg,
+        fromFieldType: field.type,
+        fromSubField,
+        aggregateOperation: AGGREGATE_OPERATIONS.percentageNotEmpty,
       };
-    }
 
-    return acc;
-  }, {});
+      if (field.type === FieldMetadataType.DATE_TIME) {
```

Comment:
I think it will be cleaner and easier to read with a switch case, you can also create something with factories for each type and a mapping class